tests/snappi_tests/multidut/ecn/test_multidut_dequeue_ecn_with_snappi.py

```diff
@@ -39,48 +37,50 @@ def test_dequeue_ecn(request,
         request (pytest fixture): pytest request object
         snappi_api (pytest fixture): SNAPPI session
         conn_graph_facts (pytest fixture): connection graph
-        fanout_graph_facts (pytest fixture): fanout graph
+        fanout_graph_facts_multidut (pytest fixture): fanout graph
         duthosts (pytest fixture): list of DUTs
+        lossless_prio_list (pytest fixture): list of all the lossless priorities
         rand_one_dut_lossless_prio (str): name of lossless priority to test, e.g., 's6100-1|3'
         line_card_choice: Line card choice to be mentioned in the variable.py file
         linecard_configuration_set : Line card classification, (min 1 or max 2  hostnames and asics to be given)
         prio_dscp_map (pytest fixture): priority vs. DSCP map (key = priority).
-
+        tbinfo (pytest fixture): fixture provides information about testbed
+        get_snappi_ports (pytest fixture): gets snappi ports and connected DUT port info and returns as a list
     Returns:
         N/A
     """
 
-    if line_card_choice not in linecard_configuration_set.keys():
-        pytest_require(False, "Invalid line_card_choice value passed in parameter")
-
-    if (len(linecard_configuration_set[line_card_choice]['hostname']) == 2):
-        dut_list = random.sample(duthosts.frontend_nodes, 2)
-        duthost1, duthost2 = dut_list
-    elif (len(linecard_configuration_set[line_card_choice]['hostname']) == 1):
-        dut_list = [dut for dut in duthosts.frontend_nodes if
-                    linecard_configuration_set[line_card_choice]['hostname'] == [dut.hostname]]
-        duthost1, duthost2 = dut_list[0], dut_list[0]
-    else:
-        pytest_require(False, "Hostname can't be an empty list")
-
-    snappi_port_list = get_multidut_snappi_ports(line_card_choice=line_card_choice,
-                                                 line_card_info=linecard_configuration_set[line_card_choice])
-    if len(snappi_port_list) < 2:
-        pytest_require(False, "Need Minimum of 2 ports for the test")
-
-    snappi_ports = get_multidut_tgen_peer_port_set(line_card_choice, snappi_port_list, config_set, 2)
-
-    testbed_config, port_config_list, snappi_ports = snappi_dut_base_config(dut_list,
-                                                                            snappi_ports,
-                                                                            snappi_api)
-
-    _, lossless_prio = rand_one_dut_lossless_prio.split('|')
-    skip_ecn_tests(duthost1)
-    skip_ecn_tests(duthost2)
+    for testbed_subtype, rdma_ports in multidut_port_info.items():
```

Comment:
In this for loop the testbed_config, port_config will be calculated multiple times. seems only the result in last iteration will be used for the run_ecn_test?

Reply:
this wont impact